### PR TITLE
Add Destructure.AsDictionary<T>()

### LINKED ISSUE #1562: Destructure.AsDictionary<T>() to override default dictionaries-are-enumerables behavior
### Description
Instances of `Dictionary<string, T>` are destructured as objects, this is also [documented](https://github.com/serilog/serilog/wiki/Structured-Data#collections). But it's inheritors does not behave the same.

I understand that class can implement multiple `IDictionary<string, T>` so such interfaces are not handled as objects. But class can not inherit from multiple `Dictionary<string, T>` so I don't think this limitation applies here.

### Reproduction
This works:
```csharp
Serilog.Log.Logger.Information("{@Instance}", new Dictionary<string, string>
{
    {"TestKey", "TestValue"}
});
```
populating **Instance** property as `{TestKey: 'TestValue'}`

But this doesn't:
```csharp
public class MyDictionary: Dictionary<string, string> {}

Serilog.Log.Logger.Information("{@Instance}", new MyDictionary
{
    {"TestKey", "TestValue"}
});
```
and populates **Instance** with `[{Key: 'TestKey', Value: 'TestValue'}]`

### Relevant package, tooling and runtime versions
Serilog 2.10.0, .NET 5.0, Windows 10

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #1562

### CODE REVIEW

**Comment by sungam3r** [comment]
@nblumhardt If OK could you add a note about this here https://github.com/serilog/serilog/wiki/Structured-Data#collections ?

**Comment by sungam3r** [comment]
Will it be included in v3 release?

File: src/Serilog/Configuration/LoggerDestructuringConfiguration.cs [hunk 1]
```diff
@@ -97,6 +100,18 @@ public LoggerConfiguration With<TDestructuringPolicy>()
         return With(new TDestructuringPolicy());
     }
 
+    /// <summary>
+    /// When destructuring objects, allows to destructure inheritors of dictionaries.
```
Comment (nblumhardt) [inline src/Serilog/Configuration/LoggerDestructuringConfiguration.cs:104]: How about:

> Capture instances of the specified type as dictionaries. By default, only concrete instantiations of <see cref="System.Collections.Generic.Dictionary{TKey,TValue}" /> are considered dictionary-like.

?

**Review by nblumhardt — CHANGES_REQUESTED** [review]
LGTM! I think we can get this into 3.0. Just had one suggestion for XDOC phrasing.

File: src/Serilog/Configuration/LoggerDestructuringConfiguration.cs (hunk 1 above)
Comment (sungam3r) [reply src/Serilog/Configuration/LoggerDestructuringConfiguration.cs]: done

**Comment by nblumhardt** [comment]
:+1:

**Pull request merged**